code/modules/antagonists/wizard/equipment/spellbook_entries/perks.dm

```diff
@@ -0,0 +1,196 @@
+		By eating someone's heart you can increase your maximum health or decrease it but get a random mutation."
+
+/datum/spellbook_entry/perks/heart_eater/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	user.AddComponent(/datum/component/heart_eater)
+
+/datum/spellbook_entry/perks/slime_friends
+	name = "Slime Friends"
+	desc = "Slimes are your friends. \
+		Every 15 seconds you lose some nutriments and summon a random evil slime to fight on your side."
+
+/datum/spellbook_entry/perks/slime_friends/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	user.AddComponent(/datum/component/slime_friends)
+
+/datum/spellbook_entry/perks/transparence
+	name = "Transparence"
+	desc = "You become a little closer to the world of the dead. \
+		Projectiles pass through you, but you lose 25% of your health and you are hunted by a terrible curse which wants to return you to the afterlife."
+
+/datum/spellbook_entry/perks/transparence/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	user.maxHealth *= 0.75
+	user.alpha = 125
+	ADD_TRAIT(user, TRAIT_UNHITTABLE_BY_PROJECTILES, REF(src))
+	RegisterSignal(user, COMSIG_ENTER_AREA, PROC_REF(make_stalker))
+
+/datum/spellbook_entry/perks/transparence/proc/make_stalker(mob/living/carbon/human/wizard, area/new_area)
+	SIGNAL_HANDLER
+
+	if(new_area == GLOB.areas_by_type[/area/centcom/wizard_station])
+		return
+	wizard.gain_trauma(/datum/brain_trauma/magic/stalker)
+	UnregisterSignal(wizard, COMSIG_ENTER_AREA)
+
+/datum/spellbook_entry/perks/magnetism
+	name = "Magnetism"
+	desc = "You get a small gravity anomaly that orbit around you. \
+		Nearby things will be attracted to you."
+
+/datum/spellbook_entry/perks/magnetism/buy_spell(mob/living/carbon/human/user, obj/item/spellbook/book, log_buy)
+	. = ..()
+	var/atom/movable/magnitizm = new /obj/effect/wizard_magnetism(get_turf(user))
+	magnitizm.orbit(user, 20)
+
+/obj/effect/wizard_magnetism
+	name = "magnetic anomaly"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shield2"
+	/// We need to orbit around someone.
+	var/datum/weakref/owner
+
+/obj/effect/wizard_magnetism/New(loc, ...)
+	. = ..()
+	transform *= 0.4
+
+/obj/effect/wizard_magnetism/orbit(atom/new_owner, radius, clockwise, rotation_speed, rotation_segments, pre_rotation)
+	. = ..()
+	if(!isliving(new_owner))
+		return
+	owner = WEAKREF(new_owner)
+	RegisterSignal(new_owner, COMSIG_ENTER_AREA, PROC_REF(check_area))
+
+/obj/effect/wizard_magnetism/proc/check_area(mob/living/wizard, area/new_area)
+	SIGNAL_HANDLER
+
+	if(new_area == GLOB.areas_by_type[/area/centcom/wizard_station])
+		return
+	START_PROCESSING(SSprocessing, src)
+	UnregisterSignal(wizard, COMSIG_ENTER_AREA)
+
+/obj/effect/wizard_magnetism/process(seconds_per_tick)
+	if(isnull(owner))
+		stop_orbit()
+		return
+	var/mob/living/wizard = owner?.resolve()
+	if(wizard.stat == DEAD)
+		stop_orbit()
+		return
```

Comment:
Why not remove these checks from `process` and just use the `COMSIG_LIVING_DEATH` and `COMSIG_QDELETED` signals to trigger this?